### PR TITLE
add EdDSA support in SSH tunnel

### LINKED ISSUE #9523: Add EdDSA support to SSH tunnel
* [ ] Update https://github.com/airbytehq/airbyte/pull/9494 with a better test (i.e. rather than just checking that the class exists, try to parse an EdDSA key)
* Publish affected connectors:
  * Destinations:
    * [ ] clickhouse
    * [ ] mariadb
    * [ ] mssql
    * [ ] mysql
    * [ ] oracle
    * [ ] postgres
  * Sources:
    * [ ] clickhouse
    * [ ] mssql
    * [ ] oracle
    * [ ] postgres

### PR DESCRIPTION
**Pull request by edgao**
## What
As title. See https://stackoverflow.com/questions/65566138/apache-mina-sshd-ssh-client-always-prints-eddsa-provider-not-supported

## How
Add the `net.i2p.crypto:eddsa` dependency.

Confirmed that the unit test fails without this dependency.

## Recommended reading order
1. build.gradle
2. `SshTunnelTest.java`

## 🚨 User Impact 🚨
nope

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

#### Community member or Airbyter
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

---
This change is [Reviewable](https://reviewable.io/reviews/airbytehq/airbyte/9494)

### CODE REVIEW

**Comment by edgao** [comment]
@tuliren do I need to bump+publish every connector that uses SshTunnel? (quick search says: clickhouse, mariadb, mssql, mysql, oracle, postgres destinations + clickhouse, mssql, oracle, postgres sources)

also do you think it's worth writing a better test (maybe actually try to parse an eddsa key or something?)

**Comment by tuliren** [comment]
Do we know why we suddenly needs this dependency, but did not see this issue earlier? The stackoverflow thread was from one year ago. So this is not a new feature introduced into Apache mina.

**Comment by edgao** [comment]
yeah, this is just a warning. I think it means that we're not properly validating the key, but the connection is still happening. Was an easy fix so just threw it into a PR.

Added a comment on the other issue, I think the user was actually seeing errors in normalization.

**Comment by tuliren** [comment]
> @tuliren do I need to bump+publish every connector that uses SshTunnel? (quick search says: clickhouse, mariadb, mssql, mysql, oracle, postgres destinations + clickhouse, mssql, oracle, postgres sources)
> also do you think it's worth writing a better test (maybe actually try to parse an eddsa key or something?)

I think it's Yes for both questions. This is a well defined project. We can create an EPIC and delegate to GL.

**Comment by edgao** [comment]
filed https://github.com/airbytehq/airbyte/issues/9523

**Comment by tuliren** [comment]
On a second thought, since this is just a warning, what about we just merge the PR, don't publish a new version, and wait for it to be published together with other updates? Bumping all SSH connectors just to mute a warning seems to have low ROI.

**Comment by yurii-bidiuk** [comment]
Added test case to verify that EdDSA is supported. Confirmed that test is failing without `net.i2p.crypto:eddsa` dependency

**Review by mkhokh-33 — APPROVED** [review]

**Review by edgao — COMMENTED** [review]
one small comment, otherwise lgtm. I can't approve b/c this is my own pr though :joy:

merged in master to hopefully get the build passing. I saw an error with spotbugs, which looks unrelated to this change, but would like to get a green build before merging.

**Review by tuliren — APPROVED** [review]
Looks good except for what Edward commented.

File: airbyte-integrations/bases/base-java/src/test/java/io/airbyte/integrations/base/ssh/SshTunnelTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import org.apache.sshd.common.util.security.SecurityUtils;
+import org.apache.sshd.common.util.security.eddsa.EdDSASecurityProviderRegistrar;
+import org.junit.jupiter.api.Test;
+
+class SshTunnelTest {
+
+  @Test
+  public void edDsaIsSupported() throws Exception {
```
[the remark this answers is not in the document]
Comment (yurii-bidiuk) [reply airbyte-integrations/bases/base-java/src/test/java/io/airbyte/integrations/base/ssh/SshTunnelTest.java:22]: done

**Comment by yurii-bidiuk** [comment]
@tuliren based on [your comment](https://github.com/airbytehq/airbyte/issues/9523#issuecomment-1029629766), can I merge this PR without publishing affected connectors?

**Comment by yurii-bidiuk** [comment]
@edgao I fixed spotBugs issue and now all checks are green

**Comment by tuliren** [comment]
> @tuliren based on [your comment](https://github.com/airbytehq/airbyte/issues/9523#issuecomment-1029629766), can I merge this PR without publishing affected connectors?

Yes. Publication does not seem necessary.

**Pull request merged**